### DISCUSSION TITLE
Intercepting localhost requests

### DISCUSSION THREAD

**Discussion by Memexurer** [mitmproxy/mitmproxy#6920] [Intercept Everything!]
Let's say that I'm hosting a simple python http.server on port 80 - and I want to see all requests sent to that server in mitmproxy - how can I achieve that?

**Comment by mhils** [comment]
https://docs.mitmproxy.org/stable/concepts-modes/ - you can set up either a reverse proxy server or configure your client to use an HTTP proxy.

**Reply by Memexurer** [reply]
Okay, but what if I can't modify HTTP client options? 
![image](https://github.com/mitmproxy/mitmproxy/assets/34003944/59244a70-6f54-4fe6-b9d3-b8ef0035139a)
Here's the example usecase of the thing which I'm trying to do: I want to capture requests to localhost servers from curl
The web request which I sent from curl doesnt show up in mitmweb.

**Reply by rosydawn6** [reply]
You have to specify the proxy port for curl  --proxy or set a system wide setting to have all traffic go through your mitmproxy instance.

**Reply by Memexurer** [reply]
im just using curl for demonstration purposes; the real app does not support http proxies or anything like that

**Reply by angristan** [reply]
Did you find a solution?
